kernel/nativevalues.ml

```diff
@@ -814,16 +845,16 @@ let no_check_arraylength t =
 let arraylength accu vA t =
   if is_parray t then
     no_check_arraylength t
-  else accu vA t
+  else apply2 accu vA t
 
 let parray_of_array t def =
   (Obj.magic (Parray.unsafe_of_obj (Obj.repr t) def) : t)
 
-let arrayinit n (f:t->t) def =
+let arrayinit n f def =
   of_parray (Parray.init (to_uint n) (Obj.magic f) def)
```

Comment:
Does your patch make it possible to eliminate this `Obj.magic`? It seems like it would fall right in its ballpark.

Reply:
You should probably just remove it to see what happens. AFAICT it's indeed not used even by the native compiler.

Reply:
https://github.com/coq/coq/pull/16575